File: demo-android/app/src/test/java/com/example/demo/integration/context/TaskTestContext.kt
package com.example.demo.integration.context

import com.example.demo.data.model.Task
import com.example.demo.data.model.TaskPriority
import com.example.demo.data.model.TaskRequest
import com.example.demo.data.model.TaskStatus
import java.util.UUID

data class TaskUpdateRequest(val taskId: String, val request: TaskRequest)

data class TaskTestContext(
    var id: String = UUID.randomUUID().toString(),
    var title: String = randomAlphabetic(12),
    var description: String? = randomAlphabetic(12),
    var status: TaskStatus = TaskStatus.TODO,
    var priority: TaskPriority = TaskPriority.MEDIUM,
    var createdDate: String? = "2024-01-15T10:00:00.000Z",
    var updatedDate: String? = "2024-01-16T12:00:00.000Z",
) {
    fun createTaskResponse(): Task = Task(
        id = id,
        title = title,
        description = description,
        status = status,
        priority = priority,
        createdDate = createdDate,
        updatedDate = updatedDate,
    )

    fun createTaskRequest(): TaskRequest = TaskRequest(
        title = title,
        description = description,
        status = status,
        priority = priority,
    )

    fun createTaskUpdateRequest(): TaskUpdateRequest = TaskUpdateRequest(
        id,
        createTaskRequest()
    )

    private companion object {
        fun randomAlphabetic(length: Int): String = buildString(length) {
            repeat(length) {
                append(('A'..'Z').random())
            }
        }
    }
}

File: demo-ios/DemoTests/Integration/Context/TaskTestContext.swift
import Foundation
@testable import Demo

struct TaskUpdateRequest: Equatable {
    let taskId: String
    let request: TaskRequest
}

struct TaskTestContext {
    var id: String
    var title: String
    var description: String?
    var status: TaskStatus
    var priority: TaskPriority
    var createdDate: String?
    var updatedDate: String?

    init(
        id: String = UUID().uuidString,
        title: String = TaskTestContext.randomAlphabetic(12),
        description: String? = TaskTestContext.randomAlphabetic(12),
        status: TaskStatus = .todo,
        priority: TaskPriority = .medium,
        createdDate: String? = "2024-01-15T10:00:00.000Z",
        updatedDate: String? = "2024-01-16T12:00:00.000Z"
    ) {
        self.id = id
        self.title = title
        self.description = description
        self.status = status
        self.priority = priority
        self.createdDate = createdDate
        self.updatedDate = updatedDate
    }

    func createTaskResponse() -> Task {
        Task(
            id: id,
            title: title,
            description: description,
            status: status,
            priority: priority,
            createdDate: createdDate,
            updatedDate: updatedDate
        )
    }

    func createTaskRequest() -> TaskRequest {
        TaskRequest(
            title: title,
            description: description,
            status: status,
            priority: priority
        )
    }

    func createTaskUpdateRequest() -> TaskUpdateRequest {
        TaskUpdateRequest(taskId: id, request: createTaskRequest())
    }

    private static func randomAlphabetic(_ length: Int) -> String {
        String((0..<length).map { _ in Character(UnicodeScalar(Int.random(in: 65...90))!) })
    }
}
